bmlab/session.py

```diff
@@ -43,14 +43,44 @@ def get_source_file_path(session_file):
 
 def get_valid_source(path):
     # If this is a session file, we need to check whether
-    # the source file exists
+    # the source file exists and is valid.
+
     if is_session_file(path):
-        path = get_source_file_path(path)
+        source_file_path = get_source_file_path(path)
+        # If the session file is valid, but the source file
+        # does not exist, we raise a file not found error.
+        if not os.path.exists(source_file_path):
+            raise FileNotFoundError(
```

Comment:
FileNotFoundError requires three arguments (had search for this myself):

```
import errno
FileNotFoundError(errno.ENOENT, "message", "path/to/filename")
```

This is important so impose can extract the file name properly from the error object.

Reply:
Done.

Reply:
I adjusted it.